### ISSUE TITLE
Product Collection: try replacing Product Image with Cover block

### ISSUE THREAD

**Issue by kmanijak** [woothemes/woocommerce#44387]
**Background**
In the issue https://github.com/woocommerce/woocommerce/issues/44146#issuecomment-1926861368, there's a suggestion to replace Product Image with Cover block so we enable all of the Cover block features. Also, it would open new capabilities in terms of patterns creation.

@nerrad proved in a quick check there's no obvious problem with that: 
![image](https://github.com/woocommerce/woocommerce/assets/20098064/352d61e8-1151-4ce8-8785-a2d55d0d905f)

**Goal:**
- Double check there are no negative side effects of using Cover instead of Product Image
- Make sure Cover block covers Product Image capabilities
- If no problems are detected:
    - replace Product Image with Cover (+ On Sale Badge) block in patterns that include Product Collection
    - use Cover (+ On Sale Badge) as a default block in Product Collection

**Comment by paaljoachim** [comment]
It is awesome to see the quick response and followup!

**Comment by paaljoachim** [comment]
The Cover block now has [Dimensions: Add Aspect Ratio block support](https://github.com/WordPress/gutenberg/pull/56897) making it possible to create product images that are the same size.

I am also suggesting creating a new Sales Badge block which by default is added as a inner block to the Cover block as I believe it will add a greater flexibility compared to the current implementation. https://github.com/woocommerce/woocommerce/issues/44392

**Comment by nerrad** [comment]
Thanks @paaljoachim, just noting we've created the issue to track this but there's still some diligence we're working through before committing to any changes.

**Comment by Marc-pi** [comment]
yep, the aspect ratio is needed in several areas, included the Related products template / loop

**Comment by ObliviousHarmony** [comment]
At first glance this seemed like a good solution since we would be leaving maintenance of the block to WordPress. Both the Cover and Image blocks provide a number of display options, however, these blocks were tailor-made for displaying and customizing a selected image. Although the Cover block lets you use the featured image, if you misconfigure the block, it won't work correctly. The depth of customization we would need to utilize the Cover block simply does not exist.

With that said, interestingly, a lot of the display options provided by those blocks are just normal `supports` declarations. For example, [the `aspectRatio` control can be added to the Product Image block trivially](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/#dimensions). Rather than trying to wrestle the Cover block to fit, I'm going to try iterating on the Product Image block to add Cover block these features there.

Since we already have an issue for that purpose (https://github.com/woocommerce/woocommerce/issues/47924) I am going to go ahead and close this one.
